### PR TITLE
Lesson 7 homework

### PR DESCRIPTION
**Pull request by alipisano**

### CODE REVIEW

File: AliPisano-alipisano/Class_5/apartment.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+require 'pry'
+
+class Apartment #create the apartment class
+	attr_accessor :name, :footage, :bedrooms, :bathrooms, :rent, :renter #create getter and setter methods using an accessor
```
Comment (Borgaard) [inline AliPisano-alipisano/Class_5/apartment.rb]: Yep, `attr_accessor` combines the getter and setter methods.

File: AliPisano-alipisano/Lesson_7_Homework/lib/building.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+class Building #create the building class
+	attr_accessor :address, :apartments, :name #creates getter and setter attributes 
+
+	def initialize(name, address) #put in required attributes, apartments will be an array of instances apartment
+		@name = name
+		@address = address
+		@apartments = [] #set equal to an empty array
+	end
+
+	def to_s #haven't covered this yet
```
Comment (Borgaard) [inline AliPisano-alipisano/Lesson_7_Homework/lib/building.rb]: `to_s` is a method you can call on certain classes to convert to a string. it's literally `5.to_string` => `"5"`. Here you're redefining and overriding this method to suit your needs.

File: AliPisano-alipisano/Lesson_7_Homework/lib/renter.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+require 'pry'
+
+class Renter #create the renter class
+	attr_accessor :name, :credit, :salary, :pets 
+
+	def initialize(name, credit, salary) #the primary job of initialize is creating an instance, build a constructor with our required attributes
+		@name = name
+		@credit = credit #assigning the information received inside the () to instance variables of the same name
```
Comment (Borgaard) [inline AliPisano-alipisano/Lesson_7_Homework/lib/renter.rb]: Correct, and you can be even more explicit with the variable on the left side of the `=`. `@name` could be `@tenant_name`. The params/arguments inside the initialize function are the same as the values on the right side of the `=` when you're defining a variable. So `(name)` is the same as `= name`.

File: AliPisano-alipisano/Lesson_7_Homework/real_estate.rb [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+ 	puts "Where is it located?"
+ 	address = gets.strip
+ 	
+ 	building = Building.new(name, address) #create building object and pass through 
+ 	add_units_to_building(building)
+end
+
+def add_units_to_building(building) # how to create an apartment multiple times
+	puts "How many apartments do you want?"
+	numbers = gets.strip.to_i #converting to an integer so can use .times method
```
Comment (Borgaard) [inline AliPisano-alipisano/Lesson_7_Homework/real_estate.rb]: :+1:

File: AliPisano-alipisano/Lesson_7_Homework/lib/apartment.rb [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+		@rent = nil #set equal to nil because aren't required to create apartment in a building
+	end
+
+#this method is inherited out of the box when you create a new class. I can override and define it myself
+	def to_s 
+		"Apartment #{name} has #{footage} square feet."
+	end 
+
+	def not_occupied?
+		renter.nil? #Ruby convention, the same as renter == nil, self.renter == nil, @renter == nill
```
Comment (Borgaard) [inline AliPisano-alipisano/Lesson_7_Homework/lib/apartment.rb]: Sometimes these reversed boolean methods can quickly become a confusing web of back and forth `not`s. 

If you were asking whether an apartment was available in English, would you ask: "Is this apartment `occupied?`" or would you ask, "Is this apartment `not_occupied?`" The first question is a quick yes or no answer, but for most folks the second question requires a second of mental boolean handling to answer.

**Comment by Borgaard** [comment]
Each time you've resubmitted your refactored your code it's gotten better and better, @alipisano! 

One request: from `bewd_sf_10_homework` do `git pull upstream master`.    
There's a merge conflict and I can't accept this pull request until that's resolved. I suspect it's because you're submitting `.DS_Store` files, which should be automatically ignored by the `.gitignore` file Travis added a while ago. 

A `.gitignore` file will ensure Git ignores any file you put in there so you don't accidentally commit and push them. When you pull, you'll pull down the `.gitignore` file which includes `.DS_Store`, and next time you push there shouldn't be a merge conflict. Try it for me? Do it with Travis or Christine if you feel you need.

**Comment by tgaff** [comment]
@Borgaard, Travis just magically fixed this by `rm`ing the .DS_Store in this repo.  However, @alipisano I still think adding the .gitignore to your repo is worth a shot.  You might consider making a global .gitignore instead: https://help.github.com/articles/ignoring-files/#create-a-global-gitignore  

Github's recommended gitignore for OSX is at: https://github.com/github/gitignore/blob/master/Global/OSX.gitignore

Those instructions above might be too vague; if so grab one of us in office-hours/class and it should only take a minute.

**Pull request merged**